spatialdata/_core/_spatial_query.py

```diff
@@ -85,3 +89,129 @@ def _bounding_box_query_points_dict(
             requested_points[points_name] = points
 
     return requested_points
+
+
+def _bounding_box_query_image(
+    image: Union[MultiscaleSpatialImage, SpatialImage], request: BoundingBoxRequest
```

Comment:
with `__annotations__` we could replace Union with `FirstType | SecondType`. Just a comment, since if we settle with that we'd have to change it everywhere.

Reply:
I am fine with that, maybe let's open an issue re refactoring them with that and once that pr is approved we can start using the new syntax.